*: implement show full processlist

erizocosmico opened the pull request:
Closes #348

ajnavarro commented on sql/analyzer/process.go:
```diff
@@ -0,0 +1,174 @@
+			}
+
+			return plan.NewResolvedTable(n.Name(), t), nil
+		case sql.Table:
```
is this necessary? Why we cannot just get ResolvedTables?

erizocosmico replied:
Not all tables may be inside ResolvedTables (e.g. SquashedTables)

ajnavarro replied:
So, if I understood correctly, maybe the easiest solution is to move the table name to the sql.Table interface, and apply the rule just to `sql.Table`s, right?

erizocosmico replied:
Yup. Not having the name in the Table is a PITA a lot of times, and there's no reason a table can't know its name (after all, it needs to return it as the `Source` in the `Schema` method, so it must know).

ajnavarro commented on sql/analyzer/pushdown_test.go:
```diff
@@ -228,5 +228,5 @@ func TestPushdownIndexable(t *testing.T) {
 	require.NoError(err)
 
-	require.Equal(expected, result)
+	require.Equal(expected, removeProcessNodes(t, result))
```
I would prefer to add process node to expected or remove the rule to the actual analyzer. WDYT?

erizocosmico replied:
It can't be added to the expected, because the nodes have functions and they're not comparable, so I guess I can remove it from the analyzer

erizocosmico commented:
@ajnavarro Table now implements sql.Nameable again, removed the name param from ResolvedTable (because it's already on the table) and removed the rule from the analyzer instead of the result nodes

erizocosmico commented:
Also rebased

ajnavarro commented on sql/analyzer/process.go:
```diff
@@ -0,0 +1,166 @@
+				return nil, err
+			}
+
+			return plan.NewResolvedTable(t), nil
```
if the table is inside a resolved table, we are adding two resolved table nodes?

erizocosmico replied:
That's exactly what the previous case is for

erizocosmico commented on sql/analyzer/process.go:
```diff
@@ -0,0 +1,166 @@
+
+	n, err := n.TransformUp(func(n sql.Node) (sql.Node, error) {
+		switch n := n.(type) {
+		case *plan.ResolvedTable:
```
If we don't keep this, we would wrap `ResolvedTable`s in `ResolvedTable`s in the next `case`

ajnavarro replied:
then we can remove 
```
t, err := wrapTable(n.Table)
if err != nil {
	return nil, err
}
return plan.NewResolvedTable(t), nil
```
Because all the tables in resovled tables will be already wrapped, right?

erizocosmico replied:
Nope, `sql.Table` is not a node and is not reached by `TransformUp`, only via `ResolvedTable`. The second case is for `sql.Table`s that are also nodes.

mcarmonaa commented on server/context.go:
```diff
@@ -11,21 +11,22 @@ import (
 
 // SessionBuilder creates sessions given a context and a MySQL connection.
-type SessionBuilder func(*mysql.Conn) sql.Session
+type SessionBuilder func(conn *mysql.Conn, addr string) sql.Session
```
The comment above is outdated.

mcarmonaa commented on server/context.go:
```diff
@@ -35,8 +36,13 @@ type SessionManager struct {
 
 // NewSessionManager creates a SessionManager with the given ContextBuilder.
-func NewSessionManager(builder SessionBuilder, tracer opentracing.Tracer) *SessionManager {
+func NewSessionManager(
```
The comment above is outdated.

erizocosmico commented:
@mcarmonaa comments fixed

@ajnavarro removed the case for `sql.Table`. Just so this can be implemented in gitbase, I also changed the schema source validation rule to just check that is not empty.
